index.js

```diff
@@ -180,72 +180,91 @@
   //  Value :: Location
   var Value = 'Value';
 
-  //  _funcPath :: (Boolean, Array String, a) -> Nullable Function
-  function _funcPath(allowInheritedProps, path, _x) {
-    var x = _x;
+  //  Prefix :: String
+  var Prefix = 'fantasy-land/';
+
+  //  funcPath :: (Array String, a) -> Nullable Function
+  function funcPath(path, value) {
+    var x = value;
     for (var idx = 0; idx < path.length; idx += 1) {
-      var k = path[idx];
-      if (x == null || !(allowInheritedProps || has(k, x))) return null;
-      x = x[k];
+      x = x && x[path[idx]];
+      if (!x) return null;
     }
     return typeof x === 'function' ? x : null;
   }
 
-  //  funcPath :: (Array String, a) -> Nullable Function
-  function funcPath(path, x) {
-    return _funcPath(true, path, x);
-  }
-
   //  implPath :: Array String -> Nullable Function
   function implPath(path) {
-    return _funcPath(false, path, implementations);
+    var x = implementations;
+    for (var idx = 0; idx < path.length; idx += 1) {
+      x = has(path[idx], x) && x[path[idx]];
+      if (!x) return null;
+    }
+    return x;
   }
 
   //  functionName :: Function -> String
-  var functionName = 'name' in function f() {} ?
+  var functionName = 'name' in identity ?
```

Comment:
We do, but I don't like coupling this check to any one of them.

If you *really* want to get rid of `function f() {}`, I could get behind replacing it with `Function`. :)